spec/federation_messages/federation_messages_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "spec_helper"
+require "federation-testbed"
+
+describe "Generation and dispatch of federation messages" do
+  before :all do
+    @testbed_pod = FactoryGirl.build(:pod).host
+    FederationTestbed.config.configure(
+      "testbed_config" => {"host" => @testbed_pod, "method" => "http"},
+      "test_target"    => {
+        "method" => "http",
+        "host"   => "localhost:3000",
+        "user"   => "alice"
+      }
+    )
+    u = FederationTestbed::User.new
+    u.id = alice.diaspora_handle
+    u.public_key = alice.public_key
+    u.private_key = nil
+    FederationTestbed.test_data.push(u)
+    rack_app = FederationTestbed::App.new
+    stub_request(:get, Addressable::Template.new(@testbed_pod + "/.well-known/host-meta")).to_rack(rack_app)
+    stub_request(:get, Addressable::Template.new(@testbed_pod + "/webfinger?q=acct:test@" + @testbed_pod))
+      .to_rack(rack_app)
+    stub_request(:get, Addressable::Template.new(@testbed_pod + "/hcard/users/test@" + @testbed_pod))
+      .to_rack(rack_app)
+    stub_request(:post, Addressable::Template.new(@testbed_pod + "/receive/users/{guid}")).to_rack(rack_app)
+  end
+  before do
+    allow_any_instance_of(Postzord::Dispatcher::Private).to receive(:deliver_to_remote).and_call_original
+    allow_any_instance_of(Postzord::Dispatcher::Public).to receive(:deliver_to_remote).and_call_original
```

Comment:
Here diffs are provided against the present PR.
1) https://gist.github.com/cmrd-senya/37614dd5789421e0f2d9
Fails because tries to perform an outgoing call that we don't stub (the stub call commented out).
2) https://gist.github.com/cmrd-senya/5b1d01a005ff590f5de4
Comment out the lines we talk about. No fails anymore => an outgoing call doesn't performed.

Is it enough as a prove?